js/reference.js

```diff
@@ -1164,26 +1171,89 @@ var ERMrest = (function(module) {
 
         /**
          * Deletes the referenced resources.
+         * @param {Array} tuples array of tuple objects used to detect differences with data in the DB
          * @returns {Promise} A promise for a TBD result or errors.
          */
-        delete: function() {
+        delete: function(tuples) {
             try {
+                verify(tuples, "'tuples' must be specified");
+                verify(tuples.length > 0, "'tuples' must have at least one row to delete");
+
                 var defer = module._q.defer();
 
                 var config = {
                     headers: {
-                        "If-Match": this._etag
+                        "If-Match": tuples[0].page._etag
                     }
                 };
-
+                var ownReference = this;
                 this._server._http.delete(this.uri, config).then(function deleteReference(response) {
-
                     defer.resolve();
                 }, function error(response) {
-                    var error = module._responseToError(response);
-                    return defer.reject(error);
+                    var status = response.status || response.statusCode;
+                    // If 412 Precondition Failed it means that ETags don't match
+                    if (status == 412) {
+                        var MAX_TRIES = 3, numTriesLeft = MAX_TRIES;
```

Comment:
I think that's a good idea. Similar to the retry logic for 500s, I've added a delay that doubles after each retry.